### DISCUSSION TITLE
Does Cloud collect analytics for the multi-search?

### DISCUSSION THREAD

**Discussion by pochka15** [meilisearch/product#791] [Cloud Feedback]
I've set the search url to `edge.meilisearch.com` but I don't see any analytics. What can be a problem?

```ts
export const meiliClient = new MeiliSearch({
  host: env.NEXT_PUBLIC_MEILISEARCH_HOST || "", // https://edge.meilisearch.com
  apiKey: env.NEXT_PUBLIC_MEILISEARCH_API_KEY || "",
});
```

![001616@2x](https://github.com/user-attachments/assets/1a29bc31-db89-4277-bd11-5bfb69d613c2)

**Comment by macraig** [comment]
Hi @pochka15, search analytics supports multi-search. Just to check, have you followed all the steps in the [Configure search analytics](https://www.meilisearch.com/docs/learn/analytics/analytics#enable-analytics-in-the-project-overview) guide?

**Reply by pochka15** [reply]
I think so. According to the docs I have to do at least two things: 1. Enable analytics in the project overview - did it as you can see on the original screenshot. 2. Update URLs in your application - we now send requests to the `edge.meilisearch.com` 👇. Am I missing something?

![001664@2x](https://github.com/user-attachments/assets/1dc091c2-7267-42de-9fee-1a55fa1b2723)

**Reply by macraig** [reply]
@pochka15 no, that looks right. Two more things you can check:
- The API key you're using might be unknown to the edge URL (you can create a new key using edge for that)
- Your env var might not loading properly, you can make a cURL request to /multi-search to detect if your infra is not correctly set up

**Reply by pochka15** [reply]
Sorry, I still gotta ask few questions.

1. Does edge always require API keys to be re-generated once analytics get enabled?

> you can create a new key using edge for that

2. I haven't found any info on how to reset the Cloud API keys (I was trying to find examples somewhere in the [Security and Persmissions](https://www.meilisearch.com/docs/learn/security/basic_security)). Can it be made through the Cloud admin panel?

> Your env var might not loading properly

3. Is it even possible? Cause I get search the results 🤔 I've also made a request through the cURL and it gave me same results.

Sorry, I might ask some silly questions but so far I don't understand how it's supposed to be working. We were using Meilisearch Cloud without analytics enabled and it worked as expected. Then we just clicked "Enable analytics" in the WEB UI and changed the API url to `edge.meilisearch.com` in our env. It still works well just as it worked before but I still cannot see any analytics in the admin panel. It feels like I'm missing some step

**Reply by macraig** [reply]
> Sorry, I might ask some silly questions but so far I don't understand how it's supposed to be working.

They aren't silly questions! 😄 In theory, you're not missing any steps, so I'll ask you to check some steps to see if it's a setup issue or if it's a bug we need to report.

> Does edge always require API keys to be re-generated once analytics get enabled?

Regenerating the API key isn't mandatory, but we recommend replacing the current keys as a precaution to ensure compatibility with the new edge route. You can do this via the API using the edge.meilisearch.com/keys route. If you're still having the same issues after that let us know.

**Reply by pochka15** [reply]
How did I not think of doing a simple test to check whether usual (not multi-search) requests get tracked. Indeed looks like it doesn't work for multi-search requests. So let me get back to my initial question, are you sure that search analytics supports multi-search?

https://github.com/user-attachments/assets/a788e016-f714-4c57-a899-9495ae1b3c46

**Reply by macraig** [reply]
Yes, it should be supported, so we'll need to investigate the issue. The engineering team will look into and we'll get back to you.

**Reply by macraig** [reply]
@pochka15 we've discovered that although multi-searches are tracked, multi-searches that are part of a federated search aren't. We're currently working on a fix, I'll let you know once it's merged. Thank you for your patience on this!

**Reply by pochka15** [reply]
@macraig Thanks for letting me know

**Reply by macraig** [reply]
@pochka15 the fix was pushed to production, you should be able to see analytics for federated searches in the "All indexes" tab
